### PR TITLE
Add --dry-run option to simplify CI system integrations

### PR DESCRIPTION
**Pull request by davido**
Fixes: #105.

Test Plan:

  $ git show --name-only HEAD | grep java$ | xargs -r gjf --dry-run
  Need Formatting:
    [gerrit-common/src/main/java/com/google/gerrit/common/IoUtil.java,\
     gerrit-server/src/main/java/com/google/gerrit/server/account/Emails.java]

### CODE REVIEW

**Comment by jbduncan** [comment]
@davido I believe this is a duplicate of https://github.com/google/google-java-format/pull/106.

**Comment by davido** [comment]
@jbduncan Oh, indeed, I missed it.

**Comment by jbduncan** [comment]
You're very welcome! :)

**Comment by sormuras** [comment]
And still waiting for an reply from the Google guys. The jitpack solution is a good work-around, though.

**Comment by dpursehouse** [comment]
@cushon any chance to get this or #106 merged?  Thanks!

**Comment by davido** [comment]
Interestingly, the buildifier tool, that is another Google's open source tool for formatting of Blaze and Bazel files, offers the `--mode=check|diff|fix` option: [1]. Unfortunately, the return code doesn't reflect the outcome of the check: [2].

* [1] https://github.com/bazelbuild/buildtools/blob/master/buildifier/buildifier.go#L44
* [2] https://github.com/bazelbuild/buildtools/issues/110

File: core/src/main/java/com/google/googlejavaformat/java/UsageException.java [hunk 1]
```diff
@@ -32,6 +32,8 @@
     "Usage: google-java-format [options] file(s)",
     "",
     "Options:",
+    "  -n, --dry-run",
+    "    Don't change anything, only check; supposed to be used form the CI.",
```
Comment (davido) [inline core/src/main/java/com/google/googlejavaformat/java/UsageException.java]: supposed to be used **from** the CI

**Comment by davido** [comment]
Actually, different check modes are needed: `fix|check|diff`. This is what Googel's buildifier is offering, so that a CI could run `-mode=check` first, to see if there were any format issues, and re-run it again with diff mode, to be more verbose on what is actually wrong. That way, tools, that support online editing, like Gerrit Code Review, could address the offending format issues and re-upload new patch set for verification. All this in browser.

Example for this approach could be seen in Tensorflow's `ci_sanity.sh` script.

https://github.com/tensorflow/tensorflow/blob/ce717be4a2966a268e9a4831c39254b611f1c9e7/tensorflow/tools/ci_build/ci_sanity.sh#L264-L294

**Comment by cushon** [comment]
#106 has been merged.

**Pull request closed without merging**